src/triedbmut.rs

```diff
@@ -31,16 +32,18 @@ pub(crate) struct TrieDBMut<'a> {
     db: &'a mut dyn HashDB,
     // When Trie is empty, root has None.
     root: &'a mut H256,
+    cache: LruCache<H256, Vec<u8>>,
 }
 
 impl<'a> TrieDBMut<'a> {
     /// Create a new trie with backing database `db` and empty `root`.
     pub fn new(db: &'a mut dyn HashDB, root: &'a mut H256) -> Self {
         *root = BLAKE_NULL_RLP;
-
+        let cache: LruCache<H256, Vec<u8>> = LruCache::new(3000);
```

Comment:
There is no change in performance when writing the value to the trie, so I will exclude the cache from this.